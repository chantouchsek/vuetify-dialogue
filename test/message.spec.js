import Vue from 'vue'
import Vuetify from 'vuetify/lib'
import { sleep } from './utils'
import VuetifyDialogPlugin from '../src'
Vue.use(Vuetify)
const vuetify = new Vuetify({})

describe('manager', () => {
  let manager

  beforeAll(() => {
    Vue.use(VuetifyDialogPlugin, { context: { vuetify } })
    manager = Vue.prototype.$dialog
  })

  it('Should not raise exception when use vuetify-dialog several times', () => {
    Vue.use(VuetifyDialogPlugin)
    VuetifyDialogPlugin.install(Vue, {})
  })

  test('Test success notification', async () => {
    const dlg = await manager.message.success('Test message', {
      timeout: 100,
      waitForResult: false
    })
    await sleep(50)
    expect(dlg.vm.$el).toMatchSnapshot()
    await sleep(600)
    await Vue.nextTick()
    expect(document.body.innerHTML).toBe('')
  })

  test('Test info notification', async () => {
    const dlg = await manager.message.info('Test message', {
      timeout: 100,
      waitForResult: false
    })
    await sleep(50)
    expect(dlg.vm.$el).toMatchSnapshot()
    await sleep(600)
    await Vue.nextTick()
    expect(document.body.innerHTML).toBe('')
  })

  test('Test error notification with top-left position', async () => {
    const dlg = await manager.message.error('Test message', {
      timeout: 100,
      position: 'top-left',
      waitForResult: false
    })
    await sleep(50)
    expect(dlg.vm.$el).toMatchSnapshot()
    await sleep(600)
    await Vue.nextTick()
    expect(document.body.innerHTML).toBe('')
  })

  test('Test warning notification with bottom-right position', async () => {
    const dlg = await manager.message.warning('Test message', {
      timeout: 100,
      position: 'bottom-right',
      waitForResult: false
    })
    await sleep(50)
    expect(dlg.vm.$el).toMatchSnapshot()
    await sleep(600)
    await Vue.nextTick()
    expect(document.body.innerHTML).toBe('')
  })
})
