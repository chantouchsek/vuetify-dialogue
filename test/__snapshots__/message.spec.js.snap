// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`manager Test error notification with top-left position 1`] = `
<div
  class="v-snack application v-snack--active v-snack--has-background v-snack--left v-snack--top"
  style="padding-bottom: 0px; padding-top: 0px;"
>
  <div
    class="v-snack__wrapper v-sheet theme--dark error"
    style=""
  >
    <div
      aria-live="polite"
      class="v-snack__content"
      role="status"
    >
      <div>
        Test message
      </div>
       
    </div>
    <div
      class="v-snack__action "
    >
      <!---->
    </div>
  </div>
</div>
`;

exports[`manager Test info notification 1`] = `
<div
  class="v-snack application v-snack--active v-snack--bottom v-snack--has-background"
  style="padding-bottom: 0px; padding-top: 0px;"
>
  <div
    class="v-snack__wrapper v-sheet theme--dark info"
    style=""
  >
    <div
      aria-live="polite"
      class="v-snack__content"
      role="status"
    >
      <div>
        Test message
      </div>
       
    </div>
    <div
      class="v-snack__action "
    >
      <!---->
    </div>
  </div>
</div>
`;

exports[`manager Test success notification 1`] = `
<div
  class="v-snack application v-snack--active v-snack--bottom v-snack--has-background"
  style="padding-bottom: 0px; padding-top: 0px;"
>
  <div
    class="v-snack__wrapper v-sheet theme--dark success"
    style=""
  >
    <div
      aria-live="polite"
      class="v-snack__content"
      role="status"
    >
      <div>
        Test message
      </div>
       
    </div>
    <div
      class="v-snack__action "
    >
      <!---->
    </div>
  </div>
</div>
`;

exports[`manager Test warning notification with bottom-right position 1`] = `
<div
  class="v-snack application v-snack--active v-snack--bottom v-snack--has-background v-snack--right"
  style="padding-bottom: 0px; padding-top: 0px;"
>
  <div
    class="v-snack__wrapper v-sheet theme--dark warning"
    style=""
  >
    <div
      aria-live="polite"
      class="v-snack__content"
      role="status"
    >
      <div>
        Test message
      </div>
       
    </div>
    <div
      class="v-snack__action "
    >
      <!---->
    </div>
  </div>
</div>
`;
